dev_tools/check_shellscript_set_options.py

```diff
@@ -21,14 +21,19 @@ def _is_valid_shell_file(filename: Path, expected_options: str) -> bool:
     return any(_sets_options_or_is_nolint(line, expected_options) for line in lines)
 
 
+def _get_possible_shebangs(program: str) -> tuple[str, ...]:
+    shebang_possibilities = ("#!/bin/{}", "#!/usr/bin/{}", "#!/usr/bin/env {}")
+    return tuple(shebang.format(program) for shebang in shebang_possibilities)
```

Comment:
Why not using a regex here, e.g. `#!/.*{program}`?

Reply:
I'm afraid of false matches. It seems [nobody is certain what's out there](https://stackoverflow.com/q/10376206/7260972), so I preferred to be specific.

Reply:
I took the idea from https://training.aspect.build/webinar-build-file-generation/query-api

Reply:
Ok, [ruff](https://github.com/astral-sh/ruff/blob/c970b794d01b6a1f943d644757863538964ea1f1/crates/ruff_linter/src/comments/shebang.rs#L12) and [pre-commit identify](https://github.com/pre-commit/identify/blob/856f0162b1bdb01e47bd31c2cf1786a8ea41bcc5/identify/identify.py#L178) also just use `#!` for shebang; let's do that too. 

[Similar to pre-commit](https://github.com/pre-commit/pre-commit/blob/7b88c63ae691cb243c3137bce8fb870523e0a884/pre_commit/parse_shebang.py#L7), we could depend on identify's implementation here. `identify` itself has no deps and an MIT license, so it should be an ok dependency for us. WDYT, @hofbi?

Reply:
Yes let's do that.

Reply:
Alternatively, you could also pin an older version of identify that still works on Python 3.8, but might not be worth the effort. I like your approach with the issue and todo note.

Reply:
Agreed, added 773e804